src/explore-education-statistics-admin/src/components/editable/ReorderableAccordion.tsx

```diff
@@ -0,0 +1,128 @@
+import useToggle from '@common/hooks/useToggle';
+import { OmitStrict } from '@common/types';
+import reorder from '@common/utils/reorder';
+import classNames from 'classnames';
+import React, {
+  cloneElement,
+  isValidElement,
+  ReactElement,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { DragDropContext, Droppable, DropResult } from 'react-beautiful-dnd';
+import styles from './ReorderableAccordion.module.scss';
+import {
+  DraggableAccordionSectionProps,
+  ReorderableAccordionSectionProps,
+} from './ReorderableAccordionSection';
+
+export interface ReorderableAccordionProps
+  extends OmitStrict<AccordionProps, 'openAll'> {
+  heading?: string;
+  onReorder: (sectionIds: string[]) => void;
+}
+
+const ReorderableAccordion = (props: ReorderableAccordionProps) => {
+  const { children, id, heading, onReorder } = props;
+
+  const [isReordering, toggleReordering] = useToggle(false);
+
+  const [sections, setSections] = useState<
+    ReactElement<ReorderableAccordionSectionProps>[]
+  >([]);
+
+  useEffect(() => {
+    const nextSections = React.Children.toArray(children).filter(
+      isValidElement,
+    ) as ReactElement<ReorderableAccordionSectionProps>[];
+
+    setSections(nextSections);
+  }, [children, id, isReordering]);
+
+  const saveOrder = useCallback(async () => {
+    if (onReorder) {
+      await onReorder(sections.map(section => section.props.id));
+      toggleReordering.off();
+    }
+  }, [onReorder, sections, toggleReordering]);
+
+  const handleDragEnd = useCallback(
+    ({ source, destination }: DropResult) => {
+      if (source && destination) {
+        setSections(reorder(sections, source.index, destination.index));
+      }
+    },
+    [sections],
+  );
+
+  const accordion = useMemo(() => {
+    return (
+      <Accordion {...props} openAll={isReordering ? false : undefined}>
+        {sections.map((child, index) => {
+          const section = child as ReactElement<
+            ReorderableAccordionSectionProps & DraggableAccordionSectionProps
+          >;
+
+          return cloneElement(section, {
+            index,
+            isReordering,
+          });
+        })}
+      </Accordion>
+    );
+  }, [isReordering, props, sections]);
+
+  return (
+    <div className={styles.container}>
+      <h2 className={classNames('govuk-heading-l', styles.heading)}>
+        {heading}
+
+        {sections.length > 1 &&
+          (!isReordering ? (
+            <Button
+              variant="secondary"
+              className="govuk-!-font-size-16 govuk-!-margin-bottom-0"
+              onClick={toggleReordering.on}
+            >
+              Reorder<span className="govuk-visually-hidden"> data files</span>
+            </Button>
+          ) : (
+            <Button
+              className="govuk-!-font-size-16 govuk-!-margin-bottom-0"
+              onClick={saveOrder}
+            >
+              Save order
+            </Button>
+          ))}
```

Comment:
Buttons like this should not be in the heading as it's not accessible. This should be placed below.

Reply:
Discussed this with Nick on Slack. The buttons have been moved out of the `h2` and some styling added to ensure things still look the right. Since the original code was copied from `EditableAccordion`, I've also made the same change there.